src/commands/lambda/flare.ts

```diff
@@ -1,14 +1,194 @@
+import {API_KEY_ENV_VAR, AWS_DEFAULT_REGION_ENV_VAR, CI_API_KEY_ENV_VAR} from './constants'
+import {getAWSCredentials, getLambdaFunctionConfig} from './functions/commons'
+import {requestAWSCredentials} from './prompt'
+import {renderError, renderLambdaFlareHeader, renderSoftWarning} from './renderers/flare-renderer'
 
 export class LambdaFlareCommand extends Command {
   private isDryRun = false
+  private functionName?: string
+  private region?: string
+  private apiKey?: string
+  private caseId?: string
+  private email?: string
+  private credentials?: AwsCredentialIdentity
 
+  /**
+   * @returns 0 if the command ran successfully, 1 otherwise.
+   */
   public async execute() {
     this.context.stdout.write(renderLambdaFlareHeader(this.isDryRun))
+
+    // Validate function name
+    let errorFound = false
+    if (this.functionName === undefined) {
+      this.context.stderr.write(renderError('No function name specified. [-f,--function]'))
+      errorFound = true
+    }
+
+    // Validate region
+    if (
+      this.region === undefined &&
+      this.functionName !== undefined &&
+      this.functionName.startsWith('arn:aws:lambda')
+    ) {
+      this.region = this.functionName.split(':')[3]
+    }
+    this.region = this.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
+    if (this.region === undefined) {
+      this.context.stderr.write(renderError('No region specified. [-r,--region]'))
+      errorFound = true
+    }
+
+    // Validate Datadog API key
+    this.apiKey = this.apiKey ?? process.env[CI_API_KEY_ENV_VAR] ?? process.env[API_KEY_ENV_VAR]
+    if (this.apiKey === undefined) {
+      this.context.stderr.write(renderError('No Datadog API key specified. [--api-key]'))
+      errorFound = true
+    }
+
+    // Validate case ID
+    if (this.caseId === undefined) {
+      this.context.stderr.write(renderError('No case ID specified. [-c,--case-id]'))
+      errorFound = true
+    }
+
+    // Validate email
+    if (this.email === undefined) {
+      this.context.stderr.write(renderError('No email specified. [-e,--email]'))
+      errorFound = true
+    }
+
+    if (errorFound) {
+      return 1
+    }
+
+    // Get AWS credentials
+    this.context.stdout.write('\n🔑 Getting AWS credentials...\n')
+    const credentials = await getAWSCredentials()
+    if (credentials === undefined) {
+      this.context.stdout.write(renderSoftWarning("No AWS credentials found, let's set them up!"))
+      await requestAWSCredentials()
+    } else {
+      this.credentials = credentials
+    }
+
+    // Get Lambda function configuration
+    this.context.stdout.write('🔍 Getting Lambda function configuration...\n')
+    const lambdaClientConfig: LambdaClientConfig = {
+      region: this.region,
+      credentials: this.credentials,
+    }
+    const lambdaClient = new LambdaClient(lambdaClientConfig)
+    if (this.functionName === undefined) {
+      throw new Error('Function name is undefined')
+    }
```

Comment:
```suggestion
    this.context.stdout.write('🔍 Getting Lambda function configuration...\n')
    const lambdaClientConfig: LambdaClientConfig = {
      region: this.region,
      credentials: this.credentials,
    }
    const lambdaClient = new LambdaClient(lambdaClientConfig)
    if (this.functionName === undefined) {
      throw new Error('Function name is undefined')
    }
```

Why don't we first check if `this.functionName` is undefined to render the error?

Also, don't throw errors. Instead, do a `stderr` write.